### PR TITLE
[WebProfilerBundle][TwigBundle] Compile assets

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | maybe #29194 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

This is a quick POC how we can extract pure CSS/JS and compile a distribution, to be committed and used by the components.

The idea is to introduce somewhat of codestyle and modern stuff. I followed the webpack-encore-bundle setup in a new local `assets/` folder (without the composer constraint). cc @weaverryan this was fun :+1: 

The goal is  to provide a single source of truth and ease maintenance. And eventually allows us to move forward in a more modern way.

I've discussed some ideas with @javiereguiluz, but for now im aiming preserve BC fully and focus on the inventory first, sort of speak.

The missing step here is still a simple binary to copy the build files into the components under src/, e.g.

```
# compile profiler, toolbar, exception-page, welcome-page
$ assets/compile
```

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "lockfileVersion": 1
```
Comment (stof) [inline package-lock.json]: This package-lock.json looks broken, as it has not dependencies in it.

**Review by stof — COMMENTED** [review]
If we go that way, the `assets` folder should be ignored in the archive (through .gitattributes) as it should be purely a contributor thing (the bundles should use the final version).

File: package-lock.json (hunk 1 above)
Comment (ro0NL) [reply package-lock.json]: fixed. I actually used `yarn install` :man_shrugging:

**Comment by ro0NL** [comment]
The new approach is ready to EOL the old one (i think we should). In theory we can now move everything per-case (ie. this PR can be merged as-is) and is 100% BC (ive only updated tabs in the profiler for now).

**Comment by javiereguiluz** [comment]
Roland, thanks for working on this change. To give more context: as you said, during the SymfonyCon we talked about this. The current situation is not ideal (we have duplicated CSS/JS assets, the JS code is defined inside Twig code, etc.)

This is slowing us down. It's not easy to do improvements in the toolbar or profile, and we're thinking about some cool changes to make them more useful (think about things like the dynamic filter added by Roland in 4.2).

But don't worry about this change. We're not going to do useless changes and we're not going to complicate things for the community. We're very focused on improving things to provide more value to end users. Thanks!

**Comment by ro0NL** [comment]
> But don't worry about this change. We're not going to do useless changes and we're not going to complicate things for the community. 

Agree. Im striving for the opposite here long-term. That said, i do like to introduce some technical/conceptual changes with this PR, in an effort to avoid never ending iterations.

- we should scope by default, instead of relying globals to be defined by SF.

```css
--sf-some: x; 
.sf-hidden {} 
.sf-component .local {}
```

- we should theme by default:

```css
// component.css requires globals.css

--sf-component-some: --var(--sf-global);

.sf-theme-dark {
   --sf-component-some: --var(--sf-dark);
}

.sf-component {
  .local {
  }
}
```

- keep the global definition small (only color/size vars + .sf-hidden like utils)
- unify duplicate concepts (i think we have n types of toggle implementations, table styles etc)

**Comment by nicolas-grekas** [comment]
Shall we close here?

**Comment by ro0NL** [comment]
i guess :}

**Pull request closed without merging**